block/manager.go

```diff
@@ -169,6 +172,8 @@ func NewManager(
 		return nil, err
 	}
 
+	isProposer := bytes.Equal(cmcrypto.AddressHash(genesis.Validators[0].PubKey.Bytes()), proposerAddress)
```

Comment:
[Follow Up]
Should we be referencing the genesis after we initialize the state?  Or should we only be referencing the state after load. 

imo we should not reference input variables once they have been loaded into some other struct (i.e. state). This could lead to weird edge cases in the future if the state as some custom logic that transforms the genesis information. 

We should consider whether we should be keeping the genesis in memory based on this too.